index.js

```diff
@@ -206,17 +206,31 @@ FSWatcher.prototype._isIgnored = function(path, stats) {
     /\..*\.(sw[px])$|\~$|\.subl.*\.tmp/.test(path)
   ) return true;
 
-  // create the anymatch fn if it doesn't already exist
-  this._userIgnored = this._userIgnored || anymatch(this._globIgnored
-    .concat(this.options.ignored)
-    .concat(arrify(this.options.ignored)
-      .filter(function(path) {
-        return typeof path === 'string' && !isglob(path);
-      }).map(function(path) {
-        return path + '/**/*';
-      })
-    )
-  );
+  if(!this._userIgnored){
+    var cwd = this.options.cwd;
+    var ignored;
+    if(cwd){
+      if(!this.options.ignored){
+        ignored = [];
+      } else {
+        ignored = arrify(this.options.ignored).map(function(path){
+          return isAbsolute(path) ? path : sysPath.join(cwd, path);
+        });
+      }
+    } else {
+      ignored = this.options.ignored;
```

Comment:
Couldn't this have just been done on line 211 and all the rest of these conditionals condensed to only `if (cwd && !this.options.ignored)`? Seems like it's doing the right thing overall, but the code could be refactored and simplified.